javascript/example-customer-account--metafields--js/extensions/customer-preferences/src/ProfilePreferenceExtension.js

```diff
@@ -0,0 +1,168 @@
+import {
+  BlockStack,
+  Button,
+  Card,
+  Form,
+  TextField,
+  Heading,
+  Icon,
+  InlineStack,
+  Modal,
+  extension,
+  Text,
+} from "@shopify/ui-extensions/customer-account";
+
+// [START setup-targets.extension]
+export default extension(
+  "customer-account.profile.block.render",
+  async (root, api) => {
+    const { customerId, nickName } = await getCustomerPreferences();
+
+    renderProfilePreferenceExtension(root, api, customerId, nickName);
+  },
+);
+// [END setup-targets.extension]
+
+// [START build-extension.ui]
+function renderProfilePreferenceExtension(root, api, customerId, nickName) {
+  const { i18n, ui } = api;
+
+  const setNickName = (value) => {
+    nickName = value;
+  };
+
+  const handleSubmit = async () => {
+    await setCustomerPreferences(customerId, nickName);
+
+    nickNameText.children[0].updateText(nickName);
```

Comment:
Can we put whatever is in `children[0]` into a const and access it this way?

Reply:
Updated to use `createText()` which allows invoking `update` directly on the component